Firmware/MVP/battsci.cpp

```diff
@@ -117,60 +294,60 @@ void BATTSCI_sendFrames()
       frameSum_87 += BATTSCI_writeByte( 0x40 );                                           //Never changes
       frameSum_87 += BATTSCI_writeByte( (spoofedVoltageToSend >> 1) );                    //Half Vbatt (e.g. 0x40 = d64 = 128 V)
 
-      if(LTC68042result_loCellVoltage_get() <= 30000 )
-      { //at least one cell is severely under-charged.  Disable Assist.
-        frameSum_87 += BATTSCI_writeByte( 0x11 );                                         //Battery SoC (upper byte)
-        frameSum_87 += BATTSCI_writeByte( 0x48 ); //20% SoC                               //Battery SoC (lower byte)
+      LiBCM_SoC = SoC_getBatteryStateNow_percent();
+      if(LiBCM_SoC <= 5 )
+      {
+        //at least one cell is severely under-charged.  Disable Assist.  0x11 0x48 is 20% SoC.  This if statement functionally performs a negaive recal.
+		//turns out 0x20 0x48 is ALSO 20% SoC
+        SoC_Bytes[0] = 0x20;
+        SoC_Bytes[1] = 0x48;
+
+        // Make sure SoC doesn't immediately spike back up
+        calculatedSoC = 0;
+        oldCalculatedSoC = 0;
+        temperature_Byte = 0x30;
+        SoCUpdateDelayCounter = 0;  // Setting this to 0 keeps SoC at 20% for extra frames so the IMA has extra time to be charged.
+
         debugUSB_sendChar('1');
       }
       else
-      { //all cells above 3.000 volts
+      { // all cells above 3.000 volts
 
-        // Battery is full. Disable Regen.   
-        if        (vCellWithESR_counts >= 39500) { //39500 = 3.9500 volts                                                    
-          frameSum_87 += BATTSCI_writeByte( 0x16 );                                         //Battery SoC (upper byte)
-          frameSum_87 += BATTSCI_writeByte( 0x20 ); //80% SoC                               //Battery SoC (lower byte)
-          debugUSB_sendChar('8');
-          //JTS2doNow: Change SoC to 81%
+        // The loop begins at 200 so if we need to we can force an SoC value to be held longer, such as if we need to do a positive or negative recal.
+        if (SoCUpdateDelayCounter >= (SoCUpdateDelayFrameIncrement + 200)) {
+          BATTSCI_calculateSoC();
+          SoCUpdateDelayCounter = 200;
+	    }
 
-        // Regen & Assist, no background charge   
-        } else if (vCellWithESR_counts >= 37000) { //37000 = 3.7000 volts                                               
```

Comment:
I'm confident the SoC algorithm I wrote last month works well.  As soon as we get your changes pulled into main, I'm going to replace this entire section with LiBCM's SoC_getBatteryStateNow_percent() result.